engine/profile_tactic.ml

```diff
@@ -129,48 +129,83 @@ let format_ratio x = (Printf.sprintf "%.1f%%" (100. *. x))
 let padl n s = ws (max 0 (n - utf8_length s)) ++ str s
 let padr_with c n s =
   let ulength = utf8_length s in
-  str (utf8_sub s 0 n) ++ str (String.make (max 0 (n - ulength)) c)
+  if Int.equal n ulength then str s
+  else if n < ulength then str (utf8_sub s 0 n)
+  else str s ++ str (String.make (n - ulength) c)
 
-let rec list_iter_is_last f = function
+let rec list_map_is_last f = function
   | []      -> []
   | [x]     -> [f true x]
-  | x :: xs -> f false x :: list_iter_is_last f xs
+  | x :: xs -> f false x :: list_map_is_last f xs
 
-let header =
+let repeat_str n s =
+  if String.is_empty s then s
+  else
+    let len = String.length s in
+    String.init (n * len) (fun i -> s.[i mod len])
+
+let header name_width =
   str " tactic                                   local  total   calls       max " ++
```

Comment:
This needs to be adjusted to `name_width`, right?

Reply:
changed